src/app/client/src/app/modules/groups/components/activity/add-activity-content-types/add-activity-content-types.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, OnInit, AfterViewInit } from '@angular/core';
+import { GroupsService } from '../../../services';
+import { ResourceService, ToasterService, NavigationHelperService } from '@sunbird/shared';
+import * as _ from 'lodash-es';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ADD_ACTIVITY_TO_GROUP, IGroupData } from '../../../interfaces';
+import { CsGroupAddableBloc } from '@project-sunbird/client-services/blocs';
+import { CsGroupSupportedActivitiesFormField } from '@project-sunbird/client-services/services/group/interface';
+import { TelemetryService, IImpressionEventInput } from '@sunbird/telemetry';
+
+
+@Component({
+  selector: 'app-add-activity-content-types',
+  templateUrl: './add-activity-content-types.component.html',
+  styleUrls: ['./add-activity-content-types.component.scss']
+})
+export class AddActivityContentTypesComponent implements OnInit, AfterViewInit {
+
+  public supportedActivityList;
+  public groupData: IGroupData;
+  private csGroupAddableBloc: CsGroupAddableBloc;
+  telemetryImpression: IImpressionEventInput;
+
+  constructor(
+    public groupService: GroupsService,
+    public resourceService: ResourceService,
+    public toasterService: ToasterService,
+    public activatedRoute: ActivatedRoute,
+    public router: Router,
+    public navigationHelperService: NavigationHelperService,
+    private telemetryService: TelemetryService
+  ) {
+    this.csGroupAddableBloc = CsGroupAddableBloc.instance;
+   }
+
+  ngOnInit() {
+    this.navigationHelperService.setNavigationUrl();
+    if (!this.csGroupAddableBloc.initialised) {
+      this.csGroupAddableBloc.init();
+    }
+    this.fetchActivityList();
+    this.activatedRoute.queryParams.subscribe((params) => {
+     this.groupData['name'] = _.get(params, 'groupName');
```

Comment:
Why are we doing this object mapping like this?

Reply:
modified the code.